### PR TITLE
week 3 assessment completed

### PR DESCRIPTION
**Pull request by OnyxFire1**
# Submitting Week 2 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Interview Questions
- [x] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by OnyxFire1** [comment]
I had trouble with creating the test for #2 and #3 to get a pass. I did not complete either one but will try to get it done before the next assessment, if able. This is just a continuation of my struggles grasping the information.

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +18,126 @@
...
+
+// const fibSeq = (length) => {
+//     let newArray = [1,1]
+//     for (let i=2; i<length; i++){
+//         newArray.push(newArray[i-2]+newArray[i-1])
+//     }
+//     return newArray
+
+// }
+// //PASS ./code-challenges.test.js
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:68]: This is excellent! It's very clear what is taking place. Good work!

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Submit some code for code challenges #2 and #3 and I will finish reviewing your PR! Great work on #1. Keep using your ability to problem solve and you will get this!

File: code-challenges.test.js [hunk 2]
```diff
@@ -79,65 +79,80 @@ const { array } = require("yargs")
...
+// Create a function that returns only accumulating sum.
+// Empty array should return empty.
 
-Good fail -  ReferenceError: emptyArr is not defined
+//Good fail -  ReferenceError: emptyArr is not defined
 
 // b) Create the function that makes the test pass.
+
+const emptyArr = (arr) => {
+    return arr.map((sum => value => sum += value)(0));
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js]: You are missing pseudo code for how this function works.

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Alright looks good glad you got some time with your mentor. Submit Pseudo code for #3 and we'll be good to go

File: code-challenges.test.js [hunk 3]
```diff
@@ -150,6 +149,19 @@ const numbersToAdd3 = []
...
+
+// PuesdoCode
+
+// Requires a function to run for each element.
+// Call a function once for each element using .map
+//Add the value of and assign the result to the variable
+//Sum will return 0 value because the array is empty
+// Return a new array.
+
+// I'm not sure if this makes sense, trying to explain it the best way I can. I did get help from my tutor.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:161]: Sounds good

**Review by WalkerTexas-coder — APPROVED** [review]

**Pull request merged**